src/passa/markers.py

```diff
@@ -2,9 +2,18 @@
 
 from __future__ import absolute_import, unicode_literals
 
+import itertools
+import operator
+
+import vistir
+
+from packaging.specifiers import SpecifierSet, Specifier
 from packaging.markers import Marker
 
 
+PYTHON_BOUNDARIES = {2: 7, 3: 9}
```

Comment:
Is this check really needed? I feel we can just always do `release[1] + 1`.

Reply:
I feel quite strongly that we should define actual boundaries; our current marker implementation produces outputs like a boundary at python 4, which obviously is pointless.

Having strongly defined boundaries allows for consolidation at the boundary point.  This means that I can accumulate all of the `<=` and `>=` specifiers into `<` and `>` specifiers as long as they sit at or below the boundaries.

It allows cleaner markers.  There is no reason not to do it

Reply:
More importantly if we _dont_ do it it means there is actually no point in doing anything at all here, because the entire purpose of adding or subtracting one is to consolidate.  If you don't define a boundary you don't consolidate anything and wind up just putting invalid python versions in people's markers where they only had valid ones, so you are just making things worse

Reply:
What should happen if the specifier goes out of bounds? What should `<=2.7` be normalised to? `<3.0`?

Reply:
The key is to just never modify a specifier to make it go out of bounds. <=2.7 stays as is